### PR TITLE
DOC: understanding code and getting started section to dev doc

### PR DESCRIPTION
**Pull request by rehassachdeva**
As noted in Issue #6996. have added the mentioned section to development_environment.rst.
Please review.
Thanks

### CODE REVIEW

File: doc/source/dev/development_environment.rst [hunk 1]
```diff
@@ -203,3 +203,19 @@ typically packaged as ``python-dbg``) is highly recommended.
...
 
 .. _Waf: https://code.google.com/p/waf/
+
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list or on the 
+NumPy Github page. It is perfectly okay if your pull requests aren't perfect,
```
Comment (charris) [inline doc/source/dev/development_environment.rst]: Hmm, the github page is for PR and bug reports, guestions should go to the mailing list.

File: doc/source/dev/development_environment.rst [hunk 2]
```diff
@@ -203,3 +203,19 @@ typically packaged as ``python-dbg``) is highly recommended.
...
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list or on the 
+NumPy Github page. It is perfectly okay if your pull requests aren't perfect, 
+the community is always happy to help. As a volunteer project, things do 
+sometimes get dropped and it's totally fine to ping us if something has sat 
+without a response for, say, a week or so.
```
Comment (charris) [inline doc/source/dev/development_environment.rst]: More like two-four weeks. Getting pinged every week it annoying.

**Comment by rehassachdeva** [comment]
Sir, have made both the changes.
Please review. Thanks

File: doc/source/dev/development_environment.rst [hunk 3]
```diff
@@ -203,3 +203,18 @@ typically packaged as ``python-dbg``) is highly recommended.
 .. _virtualenvwrapper: http://www.doughellmann.com/projects/virtualenvwrapper/
 
 .. _Waf: https://code.google.com/p/waf/
+
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list. It is perfectly okay if your pull requests aren't perfect,
```
Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: Wrap lines to 80 or 72 chars, whichever the document appears to use.

Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: Avoid using contractions. In this case the word "you're" can be removed entirely, "aren't"->"are not".

File: doc/source/dev/development_environment.rst [hunk 4]
```diff
@@ -203,3 +203,18 @@ typically packaged as ``python-dbg``) is highly recommended.
...
 .. _Waf: https://code.google.com/p/waf/
+
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list. It is perfectly okay if your pull requests aren't perfect, 
+the community is always happy to help. As a volunteer project, things do 
+sometimes get dropped and it's totally fine to ping us if something has sat
```
Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: Expand contraction.

File: doc/source/dev/development_environment.rst [hunk 5]
```diff
@@ -203,3 +203,18 @@ typically packaged as ``python-dbg``) is highly recommended.
...
+
+Understanding the code & getting started
+----------------------------------------
+
+The best strategy to better understand the code base is to pick something you 
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list. It is perfectly okay if your pull requests aren't perfect, 
+the community is always happy to help. As a volunteer project, things do 
+sometimes get dropped and it's totally fine to ping us if something has sat 
+without a response for about two-four weeks.
```
Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: two to four.

File: doc/source/dev/development_environment.rst [hunk 6]
```diff
@@ -203,3 +203,18 @@ typically packaged as ``python-dbg``) is highly recommended.
...
+
+So go ahead and pick something that annoys or confuses you about numpy, 
+experiment with the code, hang around for discussions or go through the 
+reference documents, to try to fix it. Things will fall in place and soon
```
Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: Remove comma.

Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: "Things will fall in place and..." seems a bit chatty. Perhaps something like "Given enough effort, ...".

File: doc/source/dev/development_environment.rst [hunk 7]
```diff
@@ -203,3 +203,18 @@ typically packaged as ``python-dbg``) is highly recommended.
...
+want to change and start reading the code to figure out how it works. 
+When you're in doubt, you can ask questions on the mailing list. It is perfectly okay if your pull requests aren't perfect, 
+the community is always happy to help. As a volunteer project, things do 
+sometimes get dropped and it's totally fine to ping us if something has sat 
+without a response for about two-four weeks.
+
+So go ahead and pick something that annoys or confuses you about numpy, 
+experiment with the code, hang around for discussions or go through the 
+reference documents, to try to fix it. Things will fall in place and soon 
+you'll have a pretty good understanding of the project as a whole. Good Luck!
```
Comment (madphysicist) [inline doc/source/dev/development_environment.rst]: Expand contraction.

**Comment by njsmith** [comment]
@madphysicist: what's wrong with an informal tone here? This isn't some government policy document; it's directed at (possibly intimidated) potential contributors, trying to reassure them and welcome them into the project. It's not like we avoid contractions when talking to each other...

**Comment by madphysicist** [comment]
@njsmith There is nothing wrong with an informal tone as long as all the information is there (which it is in this case). If the goal is to have an accessible and friendly intro, I rescind my comments on the basis that they were motivated by personal preference rather than any objective reasons. I am not a fan of overly complex documentation, but I do generally find it preferable to read concise and correct grammar over informal or chatty documents.

**Comment by rehassachdeva** [comment]
@njsmith @madphysicist Have updated, wrapped lines and corrected grammar, didn't change contractions. Please review. Thanks.

**Comment by madphysicist** [comment]
That looks nice and welcoming.

**Comment by njsmith** [comment]
Looks good to me -- thanks @rehassachdeva!

**Pull request merged**